pkg/ccl/changefeedccl/changefeedbase/options.go

```diff
@@ -219,6 +219,10 @@ const (
 	SinkParamSASLTokenURL           = `sasl_token_url`
 	SinkParamSASLScopes             = `sasl_scopes`
 	SinkParamSASLGrantType          = `sasl_grant_type`
+	SinkParamSASLAwsIAMRoleEnabled  = `sasl_aws_iam_role_enabled`
```

Comment:
I am definitly for it, the only concern that prevented from doing that was the fact this field is currently using a set of constants provided from sarama library: 

```go
const (
	// SASLTypeOAuth represents the SASL/OAUTHBEARER mechanism (Kafka 2.0.0+)
	SASLTypeOAuth = "OAUTHBEARER"
	// SASLTypePlaintext represents the SASL/PLAIN mechanism
	SASLTypePlaintext = "PLAIN"
	// SASLTypeSCRAMSHA256 represents the SCRAM-SHA-256 mechanism.
	SASLTypeSCRAMSHA256 = "SCRAM-SHA-256"
	// SASLTypeSCRAMSHA512 represents the SCRAM-SHA-512 mechanism.
	SASLTypeSCRAMSHA512 = "SCRAM-SHA-512"
	SASLTypeGSSAPI      = "GSSAPI"
	// SASLHandshakeV0 is v0 of the Kafka SASL handshake protocol. Client and
	// server negotiate SASL auth using opaque packets.
	SASLHandshakeV0 = int16(0)
	// SASLHandshakeV1 is v1 of the Kafka SASL handshake protocol. Client and
	// server negotiate SASL by wrapping tokens with Kafka protocol headers.
	SASLHandshakeV1 = int16(1)
	// SASLExtKeyAuth is the reserved extension key name sent as part of the
	// SASL/OAUTHBEARER initial client response
	SASLExtKeyAuth = "auth"
)
```

and in Sarama client, we still need to map `sasl_mechanism=AWS_MSK_IAM` back to `sarama.SASLTypeOAuth`, given the usage example provided by the AWS library, https://github.com/aws/aws-msk-iam-sasl-signer-go

```go
// Set the SASL/OAUTHBEARER configuration
config := sarama.NewConfig()
config.Net.SASL.Enable = true
config.Net.SASL.Mechanism = sarama.SASLTypeOAuth
config.Net.SASL.TokenProvider = &MSKAccessTokenProvider{}
...
```

I was a bit afraid that this could make the logic more complicated

Reply:
I have added an additional option for the SASL mechanism called `sasl_mechanism=AWS_IAM_ROLE` to be more generic, since the server doesn't always have to be MSK. Used the name `IAM_ROLE`, in case there is support for `IAM USER` in the future, even though using IAM user is a bad idea for most use cases.

Reply:
@cty123 you're right about having to do this behind sarama's back a bit, but i still think it would be worth it. Would you mind using `AWS_MSK_IAM`? my reasoning is as follows:

- iam auth is indeed specific to msk
- it's what amazon's sasl plugin uses (see step 8 of https://docs.aws.amazon.com/msk/latest/developerguide/create-topic.html) - 
![image](https://github.com/cockroachdb/cockroach/assets/4358545/be639226-f825-4d70-968d-5741169ca38e)
- we will want to support non-role auth in the future, and the way we'll do that is by using the same mechanism name and omitting the `sasl_aws_iam_role_arn` parameter

can you update this pr to reflect the above pls?

Reply:
I guess that makes sense, I wasn't aware of this AWS page.